src/github.com/matrix-org/dendron/dendron/main.go

```diff
@@ -67,7 +67,21 @@ func waitForSynapse(sp *proxy.SynapseProxy, synapseLog *log.Entry) error {
 func main() {
 	flag.Parse()
 
-	var synapseProxy proxy.SynapseProxy
+	proxyMetrics := prometheus.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Name: "synapse_proxy_request_durations_microseconds",
+			Help: "Histogram of microsecond durations of requests to underlying synapse for proxied requests",
+
+			// Manually curated list of expected request timings.
+			// Ranges from <1ms to <2 minutes, and then an auto-generated >2 minutes bucket.
+			Buckets: []float64{1000, 10000, 100000, 1000000, 2000000, 3000000, 4000000, 5000000, 10000000, 20000000, 30000000, 45000000, 60000000, 120000000},
```

Comment:
I am happy to add or remove buckets as we see fit (or indeed as the future of the code evolves); I pretty much was picking values I thought I'd find interesting.

I don't massively care about the range 10ms - 100ms; I can see maybe wanting one or two buckets between 100ms and 1s. Suggestions welcome.